### DISCUSSION TITLE
Block not transferring to child process when child process is whitelisted.

### DISCUSSION THREAD

**Discussion by pazho77** [tnodir/fort#277] [Q&A] [answered]
Hi,

Firstly - thank you so much for your software. It's truly excellent :)

A small issue - It seems that the "apply same rules to child process" is not being handled correctly. Simple example: a blocked program is still able to open an internet browser instance (the browser is normally whitelisted) and connect to a webpage of its own choosing. Would it be possible to patch this, so that the browser can _only_ connect to the internet if it's launched manually by me, and not if it's launched by a blocked application?

Thank you!

**Comment by tnodir** [comment, marked as answer]
1. The "Apply the same rules to child processes" option only affects new processes, so you need to restart a program after changing its options. [FAQ](https://github.com/tnodir/fort/wiki/FAQ#limitations-of-the-apply-same-rules-to-child-processes-option)

2. Let me double check this feature:
  - Add `.../Far.exe` program as Blocked and set its "Apply the same rules to child processes" option
  - Restart `Far.exe` to apply the option
  - Add `.../FireWallTest.exe` program as Allowed
  - Run `FireWallTest.exe` from `Far.exe` and check connection - Blocked
  - Allow `Far.exe` and check the running `FireWallTest.exe`'s connection - Allowed

So, it works fine per limitation 1).

Thanks for feedback!
